opm/simulators/wells/GasLiftSingleWellGeneric.cpp

```diff
@@ -805,30 +805,68 @@ reduceALQtoOilTarget_(double alq,
         auto bhp_this = getBhpWithLimit_(*bhp_opt);
         computeWellRates_(bhp_this.first, potentials);
         oil_rate = -potentials[this->oil_pos_];
-        if (oil_rate < target) {
+        water_rate = -potentials[this->water_pos_];
+        gas_rate = -potentials[this->gas_pos_];
+        if (oil_is_limited && this->controls_.hasControl(Well::ProducerCMode::ORAT) && oil_rate < this->controls_.oil_rate) {
+            limiting_control = Well::ProducerCMode::ORAT;
+            break;
+        }
+        if (oil_is_limited && this->controls_.hasControl(Well::ProducerCMode::LRAT) && (oil_rate + water_rate) < this->controls_.liquid_rate) {
+            limiting_control = Well::ProducerCMode::LRAT;
+            break;
+        }
+        if (gas_is_limited && gas_rate < this->controls_.gas_rate) {
```

Comment:
Shouldn't there be a check if the well has a GRAT control here? Similar to what is done above on line 810 and 814 for ORAT and LRAT..

Reply:
Since gas is limited we now that it has a GRAT control. I could add a assert.